python/ray/workflow/step_executor.py

```diff
@@ -146,13 +148,25 @@ def execute_workflow(workflow: "Workflow") -> "WorkflowExecutionResult":
     if workflow.executed:
         return workflow.result
     workflow_data = workflow.data
-    baked_inputs = _BakedWorkflowInputs.from_workflow_inputs(
-        workflow_data.inputs)
+
+    # If the outer workflow step skips checkpointing, it would
+    # update the context of all inner steps (except the output step).
+    step_context = workflow_context.get_workflow_step_context()
+    checkpoint_context = step_context.checkpoint_context.copy()
+    # detached := already detached or the outer step skips checkpointing
+    checkpoint_context.detached_from_dag = (
+        checkpoint_context.detached_from_dag
+        or not step_context.checkpoint_context.checkpoint)
+    # apply checkpoint context to inputs
+    with workflow_context.fork_workflow_step_context(
+            checkpoint_context=checkpoint_context):
+        baked_inputs = _BakedWorkflowInputs.from_workflow_inputs(
```

Comment:
I actually get confused by this line several times.
Is it possible for you to extract these [lines](https://github.com/ray-project/ray/blob/87760e41eec2f6a6a1936b0256170dcd4f308af6/python/ray/workflow/step_executor.py#L480-L486) and put them here 


from_workflow_inputs sounds like it's not executing the workflow. But actually, it's executing them. Put it here should make it easier to understand.

Reply:
This is not a result of this PR. I already refactored it in another PR, so I will leave a TODO here.